### PR TITLE
Use function to generate party names and other minor tweaks

### PR DESCRIPTION
**Pull request by rohanjr**
Note for `getParty()` I considered using the pattern in https://github.com/microsoft/TypeScript/issues/8419#issuecomment-216351820 but it doesn't seem to bring much benefit as you can still access `nextPartyId` with a namespace qualifier. Or I didn't understand the pattern there.

### CODE REVIEW

File: ui/src/index.test.tsx [hunk 1]
```diff
@@ -21,6 +21,24 @@ let browser: Browser | undefined = undefined;
...
+  const party = 'P' + nextPartyId.toString();
+  nextPartyId++;
+  return party;
+}
+
+test('Party names are unique', async () => {
+  let parties = new Set();
+  for (let i = 0; i < 10; i++) {
+    parties.add(getParty());
+  }
```
Comment (hurryabit) [inline ui/src/index.test.tsx]:
```suggestion
  const parties = new Set(Array(10).fill().map(() => getParty());
```

**Comment by hurryabit** [comment]
The pattern you cited is very common in JavaScript land. We definitely don't need it here.

File: ui/src/index.test.tsx [hunk 2]
```diff
@@ -21,6 +21,24 @@ let browser: Browser | undefined = undefined;
 
 let uiProc: ChildProcess | undefined = undefined;
 
+// Function to generate unique party names for us.
+// This should be replaced by the party management service once that is exposed
+// in the HTTP JSON API.
+let nextPartyId = 1;
+function getParty(): string {
```
[the remark this answers is not in the document]
Comment (rohanjr) [reply ui/src/index.test.tsx]: Not sure we really want that here, as we explicitly use strings throughout the tests for party names, e.g. going into the text input.

[the remark this answers is not in the document]
Comment (hurryabit) [reply ui/src/index.test.tsx]: Yeah, that makes sense. Forget about what I suggested. 😃

File: ui/src/index.test.tsx [hunk 3]
```diff
@@ -21,6 +21,21 @@ let browser: Browser | undefined = undefined;
...
+  const party = `P${nextPartyId}`;
+  nextPartyId++;
+  return party;
+}
+
+test('Party names are unique', async () => {
+  const parties = new Set(Array(10).fill({}).map(() => getParty()));
```
Comment (hurryabit) [inline ui/src/index.test.tsx]: `undefined` is usually a better choice for undefined values:
```suggestion
  const parties = new Set(Array(10).fill(undefined).map(() => getParty()));
```

Comment (rohanjr) [reply ui/src/index.test.tsx]: Think I liked the for loop better 😄

Comment (hurryabit) [reply ui/src/index.test.tsx]: @cocreature Please be my witness: @rohanjr prefers loops over functional code.

@rohanjr 😝

Comment (cocreature) [reply ui/src/index.test.tsx]: https://www.youtube.com/watch?v=LJvEIjRBSDA

**Pull request merged**